### ISSUE TITLE
Create automatically a link to Youtube page

### ISSUE THREAD

**Issue by kelson42** [openzim/zimfarm#477]
From the Youtube recipe online editor, this is difficult to jump on the online Youtube page. A helper in a form an automatically created link would be helpful.

**Comment by OSAMAPAR** [comment]
added 
good first issue

**Comment by asosyalface** [comment]
Parametre

**Comment by islom770** [comment]
- [ ] отлично

**Comment by kelson42** [comment]
@rgaudin  ould that be solved with a dedicated Vue.js component?

**Comment by kelson42** [comment]
@mahakporwal02 Is that something you would be interested to fix?

**Comment by mahakporwal02** [comment]
@kelson42  many `type=playlist` have more than one playlist id so should all the youtube links to the playlist be included?
For eg should this recipe include 2 links, one for each playlist?
![playlist](https://user-images.githubusercontent.com/56486682/106608953-04c7e200-658b-11eb-933b-e3e545557d1a.JPG)

**Comment by kelson42** [comment]
@rgaudin @mahakporwal02 It is true that the case where we have many comma separated values makes the thing difficult to handle. Here is what I believe would be the most adequate soluiton. We should have something like Bitwarden makes IMO.
![image](https://user-images.githubusercontent.com/1029718/106729219-8d876200-660d-11eb-9c84-62a438b091cd.png)

This allow to have multiple URL/values and handle them separatly. Because there is then only one per HTML input, it is easier then to create dynamically the link.

**Comment by rgaudin** [comment]
Agrees this would be good UI-wise.
Problem is that the whole *offliner-definition* and it's UI-usage is currently 1:1 between UI fields and data values so doing this would require some revamping of the editor that would take it outside the `good-first-issue`.

**Comment by benoit74** [comment]
I feel like we still have no clue about how to implement this correctly. Maybe once https://github.com/openzim/zimfarm/pull/1266 will be finalized we could add a "type" to fields definition and say that this is a Youtube IDs fields and needs special rendering in the UI?

Might be related to https://github.com/openzim/zimfarm/issues/891 as well

**Comment by benoit74** [comment]
@elfkuzco do you have any idea about how we could implement this?

**Comment by elfkuzco** [comment]
There's no `type` field anymore. Will they all be playlists? I reckon I could do it on the frontend side since the offliner_id is sent

**Comment by benoit74** [comment]
`id` in youtube recipes is either a list of playlists (or one single playlist) or a channel or a username.

**Comment by elfkuzco** [comment]
okay, so, i looked at the scraper and it seems to be doing an API call which is certainly not something I think is feasible on the UI. I checked out https://stackoverflow.com/questions/19795987/youtube-channel-and-playlist-id-prefixes but it seems these prefixes for determining the link type aren't always deterministic

I can write some simple code like this on the UI but in Javascript
```py
def classify_youtube_id(id):
    if id.startswith("UC"):
        return "channel"
    
    playlist_prefixes = ("PL", "RD", "UU", "LL", "FL", "WL")
    if id.startswith(playlist_prefixes):
        return "playlist"
    
    if len(id) == 11:
        return "video"
    
    return "unknown"
```
and build the link based on the classification:
- youtube.com/channel/UCabc...
- youtube.com/playlist?list=PLxyz...
- youtube.com/watch?v=dQw4...

This approach could fail and lead to links that don't work though. How do you think we should deal with it? If we had the `type`, it would have helped but I suppose it's deprecated now.

**Comment by benoit74** [comment]
Yes, `type` is deprecated. And the `type` I was referring to above is the type of the field, to say "this is a Youtube ID, use special logic for rendering" ; but you're right it does not help to known what we are dealing with in the `id` field.

And as mentioned above, on my side I have no idea about how to implement this correctly, hence my question.
